extensions/wikia/Sitemap/SpecialSitemap_body.php

```diff
@@ -69,14 +69,19 @@ public function __construct( $name = 'Sitemap' ) {
 	 * @param $subpage Mixed: subpage of SpecialPage
 	 */
 	public function execute( $subpage ) {
-		global $wgMemc, $wgRequest, $wgOut, $wgEnableSpecialSitemapExt;
+		global $wgMemc, $wgRequest, $wgOut, $wgEnableSitemapXmlExt;
 
 		if ( strpos( $subpage, '-newsitemapxml-' ) !== false ) {
-			if ( empty( $wgEnableSpecialSitemapExt ) ) {
+			if ( empty( $wgEnableSitemapXmlExt ) ) {
 				$this->print404();
 				return;
 			}
-			return ( new SpecialSitemapXmlController() )->execute( $subpage );
+
+			$wgOut->disable();
+			$response = F::app()->sendRequest( 'SitemapXml', 'index', [ 'path' => $subpage ] );
+			$response->sendHeaders();
+			echo $response->getBody();
+			return;
 		}
 
 		if ( !is_array( $wgMemc->get( wfMemcKey( 'sitemap-index' ) ) ) ) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "sitemap-index" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)